ably/realtime_presence_test.go

```diff
@@ -96,3 +104,38 @@ func TestRealtimePresence_Sync250(t *testing.T) {
 		t.Fatalf("members3: %v", err)
 	}
 }
+
+func TestRealtimePresence_EnsureChannelIsAttached(t *testing.T) {
+	beforePresenceGet := ^ably.StateEnum(0)
+	presTransitions := []ably.StateEnum{
+		ably.StateConnConnecting,
+		ably.StateConnConnected,
+		beforePresenceGet,
+		ably.StateChanAttaching,
+		ably.StateChanAttached,
+	}
+	rec := ably.NewStateRecorder(5)
+	opts := &ably.ClientOptions{
+		Listener:  rec.Channel(),
+		NoConnect: true,
+	}
+	app, client := testutil.ProvisionRealtime(nil, opts)
+	defer safeclose(t, client, app)
+	channel := client.Channels.Get("persisted:presence_fixtures")
+	if err := ably.Wait(client.Connection.Connect()); err != nil {
+		t.Fatal(err)
+	}
+	rec.Add(beforePresenceGet)
```

Comment:
Having tests that rely on such implementation details is bad IMO. We should avoid that.